text_localization_environment/ImageMasker.py

```diff
@@ -8,109 +9,63 @@ class ImageMasker:
+        xm = x0 + (x1 - x0) / 2
+        
+        cross_height = (y1 - y0) / 2
+        cross_width = (x1 - x0) / 2
 
-    def create_mask_gpu(self, base_mask, corners):
-        create_mask_kernel = cuda.cupy.ElementwiseKernel(
-            'T originalMask, raw T corners, int32 inputHeight, int32 batchCount, T fillValue',
-            'T mask',
-            '''
-                // determine our current position in the array
-                int batchIndex = i % batchCount;
-                int w = (i / batchCount) % inputHeight;
-                int h = i / batchCount / inputHeight;
-                int cornerIndex = batchIndex * 4 * 2;
+        draw.rectangle([(x0, ym - cross_height / 2), (x1, ym + cross_height / 2)], fill=color)
+        draw.rectangle([(xm - cross_width / 2, y0), (xm + cross_width / 2, y1)], fill=color)
 
-                // calculate vectors for each side of the box
-                int2 topVec = {
-                    corners[cornerIndex + 1 * 2] - corners[cornerIndex],
-                    corners[cornerIndex + 1 * 2 + 1] - corners[cornerIndex + 1]
-                };
+        return self.image
 
-                int2 bottomVec = {
-                    corners[cornerIndex + 3 * 2] - corners[cornerIndex + 2 * 2],
-                    corners[cornerIndex + 3 * 2 + 1] - corners[cornerIndex + 2 * 2 + 1]
-                };
+    def _gauss_kernel(self, kernel_size, nsig):
+        """Returns a 2D Gaussian kernel."""
 
-                int2 leftVec = {
-                    corners[cornerIndex] - corners[cornerIndex + 3 * 2],
-                    corners[cornerIndex + 1] - corners[cornerIndex + 3 * 2 + 1]
-                }; 
+        x = np.linspace(-nsig, nsig, kernel_size + 1)
+        kern1d = np.diff(st.norm.cdf(x))
+        kern2d = np.outer(kern1d, kern1d)
+        return kern2d / kern2d.sum()
 
-                int2 rightVec = {
-                    corners[cornerIndex + 2 * 2] - corners[cornerIndex + 1 * 2],
-                    corners[cornerIndex + 2 * 2 + 1] - corners[cornerIndex + 1 * 2 + 1]
-                };
+    def gauss(self, kernel_size=10, nsig=1):
+        kernel = self._gauss_kernel(kernel_size, nsig)
+        img = np.array(self.image, dtype=np.int32) / 255.0
 
-                // calculate cross product for each side of array
-                int crossTop = topVec.x * (h - corners[cornerIndex + 1]) - topVec.y * (w - corners[cornerIndex]);
-                int crossRight = rightVec.x * (h - corners[cornerIndex + 3]) - rightVec.y * (w - corners[cornerIndex + 2]);
-                int crossBottom = bottomVec.x * (h - corners[cornerIndex + 5]) - bottomVec.y * (w - corners[cornerIndex + 4]);
-                int crossLeft = leftVec.x * (h - corners[cornerIndex + 7]) - leftVec.y * (w - corners[cornerIndex + 6]);
+        for y in range(self.bbox[0][1], self.bbox[1][1]):
+            for x in range(self.bbox[0][0], self.bbox[1][0]):
+                acc = np.array([0.0, 0.0, 0.0])
```

Comment:
Why are you starting with a non-empty array (for the convoluted image section)? 🤔You probably tested the implementation, but it's not obvious to me